### ISSUE TITLE
Parse `unsafe` in attributes

### ISSUE THREAD

**Issue by Coekjan** [mozilla/cbindgen#1013]
See https://blog.rust-lang.org/2024/10/17/Rust-1.82.0.html#unsafe-attributes

Now attributes can contain `unsafe`, but cbindgen gives the following error:

```txt
ParseSyntaxError { crate_name: "...", src_path: "...", error: Error("expected identifier, found keyword `unsafe`") }
```

I think this issue could be urgent (should be done before edition 2024), because programmers, like me, may write:

```rs
#[unsafe(export_name = "xxx")]
pub fn f() {

}
```

Current rust stable (1.82.x) introduces this syntax and recommends us to add `unsafe`, while the old form (without `unsafe`) is still acceptable. But the old form will be a **hard error** in edition 2024.

---

This issue seems actually from upstream: https://github.com/dtolnay/syn/issues/1710

**Comment by Coekjan** [comment]
Because https://github.com/dtolnay/syn/issues/1710 is done, all we need is to wait for the next release of `syn`.

**Comment by qryxip** [comment]
Not yet completed, I think. The current cbindgen itself does not know unsafe attributes.

<https://github.com/mozilla/cbindgen/blob/3ed9434f097b3a05b4ed6eedf80877d5d4b9ace9/src/bindgen/utilities.rs#L196-L198>

They are now parsed indeed, but just skipped.

**Comment by Coekjan** [comment]
> Not yet completed, I think. The current cbindgen itself does not know unsafe attributes.
> 
> https://github.com/mozilla/cbindgen/blob/3ed9434f097b3a05b4ed6eedf80877d5d4b9ace9/src/bindgen/utilities.rs#L196-L198
> 
> They are now parsed indeed, but just skipped.

Yes, you are right. cbindgen should visit the inner of the `unsafe` attributes.

**Comment by emilio** [comment]
Fixed by #1020, thanks @Coekjan!

**Comment by Coekjan** [comment]
> Fixed by #1020, thanks @Coekjan!

@emilio Could you please make a new release? My project needs this feature (to parse unsafe attributes).
